core/core-api/src/main/java/io/fabric8/launcher/core/spi/Application.java

```diff
@@ -23,8 +25,19 @@
 public @interface Application {
 
     enum ApplicationType {
+
         LAUNCHER,
-        OSIO
+        OSIO;
+
+        public static ApplicationType fromHeader(HttpServletRequest request) {
```

Comment:
@gastaldi moved that method to the type itself as it's used in both app-aware producer and token validation

Reply:
The HttpServletRequest in there kinda triggers me the core with a strong dependency on the Servlet API.

Reply:
But maybe that's ok, WDYT @quintesse ?

Reply:
Yeah this is a bit of a smell. On the other hand it's where we use that thing anyway. Alternatively one can make yet another `Util` living somewhere closer to servlet world.